# Capstone 2:
## Business Analysis with Python
I recently had the chance of working with EmporiUm, a growing student bookstore chain that sells tech, books, and other school essentials. With EmporiUm, I had the opportunity of working with the marketing manager to compare Maine and Maryland territories. During the analysis, I got to review, clean, analyze, display, and make recommandations to help improve overall.

In [1]:
import numpy as np
import pandas as pd
import matplotlib as plt

### StoreSales

In [2]:
ss = pd.read_csv('StoreSales.csv')
ss.info()
ss.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 169027 entries, 0 to 169026
Data columns (total 5 columns):
 #   Column            Non-Null Count   Dtype  
---  ------            --------------   -----  
 0   Transaction Date  169027 non-null  object 
 1   Store ID          169027 non-null  int64  
 2   RewardsID         17599 non-null   float64
 3   Prod Num          169027 non-null  object 
 4   Sale Amount       169027 non-null  float64
dtypes: float64(2), int64(1), object(2)
memory usage: 6.4+ MB


,Transaction Date,Store ID,RewardsID,Prod Num,Sale Amount
0,1/1/2022,702,NaN,105349-M,8.00
1,1/1/2022,704,NaN,105350-T,144.00
2,1/1/2022,705,NaN,105351-M,44.00
3,1/1/2022,705,NaN,105352-M,47.61
4,1/1/2022,705,NaN,105353-A,20.36


### Store Details 

In [3]:
sd = pd.read_csv('StoreDetail.csv')
sd.info()
sd.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 111 entries, 0 to 110
Data columns (total 6 columns):
 #   Column             Non-Null Count  Dtype 
---  ------             --------------  ----- 
 0   Store Location     111 non-null    object
 1   State              111 non-null    object
 2   Store ID           111 non-null    int64 
 3   Territory Manager  111 non-null    object
 4   Region             111 non-null    object
 5   Region Director    111 non-null    object
dtypes: int64(1), object(5)
memory usage: 5.3+ KB


,Store Location,State,Store ID,Territory Manager,Region,Region Director
0,Aurora,Colorado,701,Jim Heck,West,Cassie Chambers
1,Berthoud,Colorado,702,Jim Heck,West,Cassie Chambers
2,Boulder,Colorado,703,Jim Heck,West,Cassie Chambers
3,Castle Rock,Colorado,704,Jim Heck,West,Cassie Chambers
4,Denver,Colorado,705,Jim Heck,West,Cassie Chambers


### Products

In [4]:
prod = pd.read_csv('Products.csv')
prod.info()
prod.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 669 entries, 0 to 668
Data columns (total 4 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   Prod Num       669 non-null    object
 1   Product        669 non-null    object
 2   CategoryID     669 non-null    int64 
 3   SubcategoryID  669 non-null    object
dtypes: int64(1), object(3)
memory usage: 21.0+ KB


,Prod Num,Product,CategoryID,SubcategoryID
0,105248-IT,TCL NXTPAPER 10s,120,120-tab
1,105249-IT,Dell Latitude 7320 Detachable,120,120-tab
2,105250-IT,Realme Pad,120,120-tab
3,105251-IT,Lenovo Tab P12 Pro,120,120-tab
4,105252-IT,Microsoft Surface Pro 9,120,120-tab


### Product Categories

In [5]:
prod_cate = pd.read_csv('ProductCategories.csv')
prod_cate.info()
prod_cate.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 52 entries, 0 to 51
Data columns (total 4 columns):
 #   Column         Non-Null Count  Dtype 
---  ------         --------------  ----- 
 0   CategoryID     52 non-null     int64 
 1   Category       52 non-null     object
 2   SubcategoryID  52 non-null     object
 3   Subcategory    52 non-null     object
dtypes: int64(1), object(3)
memory usage: 1.8+ KB


,CategoryID,Category,SubcategoryID,Subcategory
0,120,Technology & Accessories,120-tab,Tablets
1,120,Technology & Accessories,120-cal,Calculators
2,120,Technology & Accessories,120-sof,Software Download
3,120,Technology & Accessories,120-hea,Headphones
4,120,Technology & Accessories,120-ext,External Accessories


### Customer List

In [6]:
cust = pd.read_csv('customer_list.csv')
cust.info()
cust.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 521 entries, 0 to 520
Data columns (total 1 columns):
 #   Column                                           Non-Null Count  Dtype 
---  ------                                           --------------  ----- 
 0   cust_id|date|time|name|email|phone|sms-opt-out   521 non-null    object
dtypes: object(1)
memory usage: 4.2+ KB


,cust_id|date|time|name|email|phone|sms-opt-out
0,1|2023-03-15|08:45:12|Rachel|rachel@centralper...
1,2|2023-05-22|12:30:45|R. Geller|rossg@centralp...
2,3|2023-07-09|18:15:27|Monica Geller|chefmonica...
3,4|2023-09-01|21:05:33|Chandler Bing|chandlerb@...
4,5|2023-11-18|14:22:10|Joey|howyoudoing@central...


## Core Marketing Analysis
### The marketing manager wants to know:

Who are the territory managers for the sales territories assigned? 

What are the store IDs and cities for the stores in each assigned sales territory?

In [7]:
filtered = sd[sd['State'].isin(['Maine', 'Maryland'])]

grouped = filtered.groupby('Territory Manager')

for manager, data in grouped:
    print(f"\nTerritory Manager: {manager}")
    for _, row in data.iterrows():
        print(f"  Store ID: {row['Store ID']} - City: {row['Store Location']}")


Territory Manager: Erbayne Middleton
  Store ID: 818 - City: Bangor
  Store ID: 819 - City: Bar Harbor
  Store ID: 820 - City: Kennebunkport
  Store ID: 821 - City: Lewiston
  Store ID: 822 - City: Orono
  Store ID: 823 - City: South Portland

Territory Manager: Shruti Reddy
  Store ID: 731 - City: Annapolis
  Store ID: 732 - City: Back River
  Store ID: 733 - City: Baltimore
  Store ID: 734 - City: Germantown
  Store ID: 735 - City: Howard
  Store ID: 736 - City: North Harford
  Store ID: 737 - City: Parkville
  Store ID: 738 - City: Queen Anne's County
  Store ID: 739 - City: Ridgely


I went into the Store Detail csv file and saw that all the States, associated managers, with store id numbers where located there. Using the data I found I was able to deduce that the manager for Maine is Erbayne Middleton and Shruti Reddy for Maryland. I was also able to find the corresponding Store IDs for both territories. Store IDs 818 - 823 belong to Maine as Store IDs 731 - 739 belong to Maryland. 

### The Marketing Manager wants to know:

What is monthly total revenue for in-store sales in each of the two sales territories, over the full
period covered by the data?

In [16]:
filtered.head()

,Store Location,State,Store ID,Territory Manager,Region,Region Director
37,Bangor,Maine,818,Erbayne Middleton,Northeast,Michael Jarvis
38,Bar Harbor,Maine,819,Erbayne Middleton,Northeast,Michael Jarvis
39,Kennebunkport,Maine,820,Erbayne Middleton,Northeast,Michael Jarvis
40,Lewiston,Maine,821,Erbayne Middleton,Northeast,Michael Jarvis
41,Orono,Maine,822,Erbayne Middleton,Northeast,Michael Jarvis


In [17]:
ss.head()

,Transaction Date,Store ID,RewardsID,Prod Num,Sale Amount
0,1/1/2022,702,NaN,105349-M,8.00
1,1/1/2022,704,NaN,105350-T,144.00
2,1/1/2022,705,NaN,105351-M,44.00
3,1/1/2022,705,NaN,105352-M,47.61
4,1/1/2022,705,NaN,105353-A,20.36


In [57]:
df_merge = pd.merge(ss, filtered, on = 'Store ID')
df_merge

,Transaction Date,Store ID,RewardsID,Prod Num,Sale Amount,Store Location,State,Territory Manager,Region,Region Director
0,1/1/2022,731,NaN,105384-A,23.44,Annapolis,Maryland,Shruti Reddy,Northeast,Michael Jarvis
1,1/1/2022,733,20.0,105385-M,27.52,Baltimore,Maryland,Shruti Reddy,Northeast,Michael Jarvis
2,1/1/2022,734,NaN,105386-M,25.00,Germantown,Maryland,Shruti Reddy,Northeast,Michael Jarvis
3,1/1/2022,735,NaN,105349-M,8.00,Howard,Maryland,Shruti Reddy,Northeast,Michael Jarvis
4,1/1/2022,736,NaN,105310-IT,71.74,North Harford,Maryland,Shruti Reddy,Northeast,Michael Jarvis
...,...,...,...,...,...,...,...,...,...,...
56448,6/19/2024,821,NaN,105531-A,57.70,Lewiston,Maine,Erbayne Middleton,Northeast,Michael Jarvis
56449,6/19/2024,822,NaN,105316-IT,33.13,Orono,Maine,Erbayne Middleton,Northeast,Michael Jarvis
56450,6/19/2024,823,NaN,105469-M,14.60,South Portland,Maine,Erbayne Middleton,Northeast,Michael Jarvis
56451,6/19/2024,823,NaN,105848-M,33.10,South Portland,Maine,Erbayne Middleton,Northeast,Michael Jarvis


In [56]:
df_merge['Transaction Date'] = pd.to_datetime(df_merge['Transaction Date'])
df_merge['Month'] = df_merge['Transaction Date'].dt.to_period('M')

monthly_sales = df_merge.groupby(['Month', 'Store ID', 'Store Location', 'State'])['Sale Amount'].sum()

print(monthly_sales.head(30))
print(monthly_sales.tail(30))

Month    Store ID  Store Location       State   
2022-01  731       Annapolis            Maryland      3924.44
         732       Back River           Maryland      2383.74
         733       Baltimore            Maryland      2485.49
         734       Germantown           Maryland      7795.47
         735       Howard               Maryland      3691.09
         736       North Harford        Maryland    161480.45
         737       Parkville            Maryland      5028.56
         738       Queen Anne's County  Maryland      1998.79
         739       Ridgely              Maryland      1276.87
         818       Bangor               Maine         2632.18
         819       Bar Harbor           Maine         2411.10
         820       Kennebunkport        Maine         3181.89
         821       Lewiston             Maine         1260.12
         822       Orono                Maine         1665.16
         823       South Portland       Maine         4549.86
2022-02  731       An

I figured if I need to find the overall total of in store sales by month, then I would need to be able to see the sale amount, date, store, and territory. I seen that Store Sales contained the a sale amount column and a transaction date column which were already half of the checklist. I went to back to the Store Detail file because one I already had it filtered with my specific territories, and two, I already seen the other columns I needed to work with. I felt I needed to see it all on one table so it would be easier to view it so I merged them based on Store ID. I was asked to show it month by month, so I decided to present the first and last two months instead of 50,000+ rows.